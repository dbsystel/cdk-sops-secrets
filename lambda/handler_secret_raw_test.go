package main

import (
	"testing"

	"github.com/gkampitakis/go-snaps/snaps"
)

func Test_FullWorkflow_Create_S3_RAW_Simple(t *testing.T) {
	mocks, ctx, event := prepareHandler(t, "events/event_create_s3_secret_raw_simple.json")

	phys, data, err := mocks.syncSopsToSecretsmanager(ctx, event)
	check(err)
	snaps.MatchSnapshot(t, ">>>syncSopsToSecretsmanager", phys, data, err)
}
