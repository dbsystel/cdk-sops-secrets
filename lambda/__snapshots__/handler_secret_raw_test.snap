
[Test_FullWorkflow_Create_S3_RAW_Simple - 1]
>>>S3MAnagerMockClient.Download.Input
{
  Bucket: "..",
  Key: "../test-secrets/binary/sopsfile.enc-age.binary"
}
---

[Test_FullWorkflow_Create_S3_RAW_Simple - 2]
>>>SecretsManagerMockClient.PutSecretValue.Input
main.putSecretValueInputNotSecure{
    _:                  struct {}{},
    ClientRequestToken: &"../test-secrets/binary/sopsfile.enc-age",
    RotationToken:      (*string)(nil),
    SecretBinary:       nil,
    SecretId:           &"arn:aws:secretsmanager:eu-central-1:123456789012:secret:testsecret",
    SecretString:       &"-----BEGIN RSA PRIVATE KEY-----\nMIICXQIBAAKBgQDVd/OAdqlMznWINBDoyR+PESgQJlUptwnh/vzbgAaIuHl4gN7Q\nW2Jx06bKFgv8yAQ4ouR0EaPv43X1pWFUyE/vK4e2A4M28Ciriy5r2TiBx57EL7bA\nAiiCaEvpKx3uAtTgFYcdLdD5xYKUf51gymZ6wU8BBgYMkVlL5rpGmR7ddQIDAQAB\nAoGAaeEXG+6/RmNZFHeOs4eeaZ2+21PIBerNshSBYnX7x7CIP2bcHUhWadLq9W0+\nAOFMhrL00UU2pPOTPIYM4rqAOl5PvPeV4GNMvI+9/619p/ZNMbnvj43KqUtaPp1g\nJyT22CH4TqDC2io8S9xWxxh9AgvgOI/ekJ3h4PdYpmyVUXkCQQD6b1dkYhlgiy3u\nZ84L5WM+7vXG2ivA3N3T5eYJdsQFvlV6gNSqbKcxIR+cFK0RaRiWrd7U2IOM67RF\nYRvHNN0rAkEA2jZRXVjHjTZSuAVDMOA4cc6pgxOqByg14clWgRDtbueIMHNCYLJY\nbi5YE0kqzjSaH2WbeT7LX1q3S7amoN4f3wJBALw9ZrYYmrSbyRmTQyhj8raCTZF7\nujmMitzUyJVChVM/3uZm4fN8GivuluDuFaypj4brCDx6xl7taKJhvMx+quMCQF/C\nZoJoa2n05OgMpyfTvfFzl8AF6R+q7bpf+K47F3cL9CAO9JoqqdPwUoZkHXzQaLJO\njKPwgp8d2EJJrWX7FFECQQCRMPE4WsTCLXmVefZne7k000WPclZ4pIjggz1lpGgN\n23bwNOla1k4/B10btnEzNi16/b01Kf3K4hYaicd46sLH\n-----END RSA PRIVATE KEY-----",
    VersionStages:      nil,
}
---

[Test_FullWorkflow_Create_S3_RAW_Simple - 3]
>>>syncSopsToSecretsmanager
arn:custom:sopssync:eu-central-1:123456789012:secret:testsecret
map[string]interface {}{
    "ARN":           "arn:custom:sopssync:eu-central-1:123456789012:secret:testsecret",
    "Name":          "dfb70680d326c245dbdd113d635caa16113e78b1eb8b7447f10eeb861e491724",
    "VersionId":     "afe1e68ea88d603e524fc4854b2f33e8f56ce0cc80b539d21fddcbf82586fb38",
    "VersionStages": []*string{
        &"dfb70680d326c245dbdd113d635caa16113e78b1eb8b7447f10eeb861e491724",
    },
}
nil
---
